protocol/index.ts

```diff
@@ -729,6 +777,26 @@ export interface ArcjetEmailRule<Props extends { email: string }>
   allowDomainLiteral: boolean;
 }
 
+export type CustomDetect<Custom extends string> = (
+  tokens: string[],
+) => (SensitiveInfoEntity<Custom> | undefined)[];
+
+export type SensitiveInfoConfig<Custom extends string> = {
+  allow?: SensitiveInfoEntity<Custom>[];
+  deny?: SensitiveInfoEntity<Custom>[];
+  customDetect?: CustomDetect<Custom>;
+  contextWindowSize: number;
+};
+
+export interface ArcjetSensitiveInfoRule<
+  Props extends {},
+  Custom extends string,
+> extends ArcjetLocalRule<Props> {
+  type: "SENSITIVE_INFO";
+
+  options: SensitiveInfoConfig<Custom>;
```

Comment:
Removed in my PR because you don't need to hold onto the config on the rule. The properties on other rules are due to remote serialization.